flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/BigQuerySinkConfig.java

```diff
@@ -123,9 +142,115 @@ public Builder serializer(BigQueryProtoSerializer serializer) {
+        }
+
         public BigQuerySinkConfig build() {
+            if (deliveryGuarantee == DeliveryGuarantee.EXACTLY_ONCE) {
+                validateStreamExecutionEnvironment(env);
+            }
             return new BigQuerySinkConfig(
                     connectOptions, deliveryGuarantee, schemaProvider, serializer);
         }
     }
+
+    // DO NOT USE!
+    // This method is used internally to create sink config for the Table API integration.
+    // Note that the serializer is hard coded for Flink Table's RowData.
+    @Internal
+    public static BigQuerySinkConfig forTable(
+            BigQueryConnectOptions connectOptions,
+            DeliveryGuarantee deliveryGuarantee,
+            LogicalType logicalType) {
+        return new BigQuerySinkConfig(
+                connectOptions,
+                deliveryGuarantee,
+                new BigQuerySchemaProviderImpl(
+                        BigQueryTableSchemaProvider.getAvroSchemaFromLogicalSchema(logicalType)),
+                new RowDataToProtoSerializer());
+    }
+
+    public static void validateStreamExecutionEnvironment(StreamExecutionEnvironment env) {
+        if (env == null) {
+            throw new IllegalArgumentException(
+                    "Expected StreamExecutionEnvironment, found null."
+                            + " Please provide the StreamExecutionEnvironment used in Flink job.");
+        }
+        validateRestartStrategy(env.getRestartStrategy());
+    }
+
+    private static void validateRestartStrategy(RestartStrategyConfiguration restartStrategy) {
+        if (restartStrategy == null) {
+            throw new IllegalArgumentException(
+                    "Could not read RestartStrategyConfiguration from StreamExecutionEnvironment."
+                            + " Please provide the StreamExecutionEnvironment used in Flink job and"
+                            + " set a restart strategy.");
+        }
+        // Restart configurations are mostly being checked against Flink defaults for optimum
+        // interactions with external systems, primarily BigQuery storage write APIs.
+        // Keep in mind, maximum 10,000 CreateWriteStream calls are allowed by BigQuery per hour per
+        // project per region.
+        if (restartStrategy instanceof FixedDelayRestartStrategyConfiguration) {
+            FixedDelayRestartStrategyConfiguration strategy =
+                    (FixedDelayRestartStrategyConfiguration) restartStrategy;
+            if (strategy.getDelayBetweenAttemptsInterval().toMilliseconds()
+                            < MILLISECONDS_PER_SECOND
+                    || strategy.getRestartAttempts() > 10) {
+                throw new IllegalArgumentException(
+                        "Invalid restart strategy: FixedDelayRestartStrategyConfiguration should"
+                                + " be used with minimum restart delay 1 second,"
+                                + " and maximum 10 restart attempts.");
```

Comment:
lets include in this error what the values for strategy.getDelayBetweenAttemptsInterval().toMilliseconds() and strategy.getRestartAttempts() are